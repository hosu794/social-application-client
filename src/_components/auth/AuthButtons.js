import React from "react";
import { Link } from "react-router-dom";

function AuthButtons() {
  return (
    <React.Fragment>
      <Link to="/register" className="button is-primary">
        <strong>Sign up</strong>
      </Link>
      <Link to="/login" className="button is-light">
        Log in
      </Link>
    </React.Fragment>
  );
}

export default AuthButtons;
