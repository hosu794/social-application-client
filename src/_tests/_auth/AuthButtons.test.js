import React from "react";
import { MemoryRouter, Link } from "react-router-dom";
import AuthButtons from "../../_components/auth/AuthButtons";
import { mount } from "enzyme";

describe("Test for the AuthButtons", () => {
  const TestingComponent = () => (
    <MemoryRouter>
      <AuthButtons />
    </MemoryRouter>
  );

  let wrapper = mount(<TestingComponent />);

  test("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should be defined", () => {
    expect(AuthButtons).toBeDefined();
  });

  test("should first Link constains correct value", () => {
    let element = wrapper.find(Link).at(0);
    expect(element.contains(<strong>Sign up</strong>)).toEqual(true);
  });

  test("should second Link constains correct value ", () => {
    let element = wrapper.find(Link).at(1);
    expect(element.text()).toEqual("Log in");
  });

  test("should first Link have correct classes", () => {
    let element = wrapper.find(Link).at(0);
    expect(element.hasClass("button is-primary")).toEqual(true);
  });

  test("should second Link have correct classes", () => {
    let element = wrapper.find(Link).at(1);
    expect(element.hasClass("button is-light")).toEqual(true);
  });
});
