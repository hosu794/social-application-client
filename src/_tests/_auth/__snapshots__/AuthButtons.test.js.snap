// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the AuthButtons should render correctly 1`] = `ReactWrapper {}`;
